luxonis_ml/nn_archive/config_building_blocks/base_models/head.py

```diff
@@ -172,17 +211,20 @@ def __init__(self):
 class Head(CustomBaseModel):
     """Represents head of a model.
 
-    @type head_id: str
-    @ivar head_id: Unique head identifier.
+    @type outputs: C{Union[List[str], Dict[str, Union[str, List]]]}
+    @ivar outputs: A list of output names from the `outputs` block of the archive or a dictionary mapping DepthAI parser names needed for the head to output names. The referenced outputs will be used by the DepthAI parser.
     @type metadata: HeadMetadata
     @ivar metadata: Parameters required by head to run postprocessing.
     """
 
-    head_id: str = Field(description="Unique head identifier.")
+    outputs: Union[List[str], Dict[str, Union[str, List]]] = Field(
+        description="A list of output names from the `outputs` block of the archive or a dictionary mapping DepthAI parser names needed for the head to output names. The referenced outputs will be used by the DepthAI parser."
```

Comment:
I assume only the outputs relevant to the head operations must be listed here? From the perspective of automatic NN archive generation, I'm wondering how do we plan to identify those. We could try to draw it from cfg model architecture but this might not be so straight-forwardas as outputs now became the outputs of the executable (e.g. ONNX) and might differ from what we have in cfg (as far as I know).

Reply:
@jkbmrz I was considering overriding `outputs` for each `HeadMetadata` or at least providing some validation on the expected format of `outputs` for a given head. However, we would then need to define `outputs` on the `HeadMetadata` level (instead of `Head`) or have one really long validtor in `Head` (which I don't think is a good idea). If we move `outputs` to `HeadMetadata`, then we only have `HeadMetadata` attributes and no attributes specific to `Head`. 

Thus, it might make sense removing `HeadMetadata` entirely and just having all the head types inherit from `Head`. Then, we overrride `outputs` for each head, further defining its type and validating possible param names (in the `outputs` keys). Thoughts on that idea?

Reply:
We could remove the `HeadMetadata` class and have all the head types inherit from `Head` but I'm still not sure how we would figure out automatically what outputs belong to a specific head. Or does overriding `outputs` for each head means  predefining?

Reply:
Commit [2dcc2c3](https://github.com/luxonis/luxonis-ml/pull/82/commits/2dcc2c3db3253e289588603af1e65faeadbb5754) implements the idea